engines/startrek/filestream.cpp

```diff
@@ -0,0 +1,35 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "startrek/filestream.h"
+
+namespace StarTrek {
+
+FileStream::FileStream(byte *data, uint32 len, bool bigEndian) : Common::MemoryReadStreamEndian(data, len, bigEndian) {
+	_data = data;
+}
+
+FileStream::~FileStream() {
+	free(_data);
```

Comment:
Instead of doing this, I suggest to extend MemoryReadStreamEndian class constructor with the optional "disposeMemory" flag, just as the MemoryReadStream does, pass it to MemoryReadStream, and kill this FileStream class altogether.

Reply:
I can do that, but as I'm still using the "_data" field, I'd prefer to still keep this class unless you feel strongly about it.